### PR TITLE
.github: add a basic CI workflow

### PR DESCRIPTION
**Pull request by creachadair**
Check that Go code is formatted, tests pass, and lint checks are clean.

### CODE REVIEW

**Review by Erisa — APPROVED** [review]
Thank you! This looks good to me, but I'd like to wait for #1 to be merged before merging this so these checks are passing when merged in.

**Comment by creachadair** [comment]
> Thank you! This looks good to me, but I'd like to wait for #1 to be merged before merging this so these checks are passing when merged in.

Yup, that's my intent also 😄

**Comment by Erisa** [comment]
Pulled in main and it looks good, thanks again!

**Pull request merged**
